### PR TITLE
SDPA extend backward realized tensor alignment checking to forward realized tensors

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #116069

The logic to check alignment for realized tensors in the backward can be extended for realized tensors in the forward. This fixes an interaction with freezing.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_inductor/lowering.py [hunk 1]
```diff
@@ -2087,13 +2074,20 @@ def is_aligned_backward(x):
...
+            arg.get_stride()
+            if is_aligned_realized_tensor(arg):
                 return arg
+        except AttributeError:
+            pass
+
+        def is_aligned(x):
+            return (V.graph.sizevars.size_hint(x.get_size()[-1]) % ALIGNMENT) == 0
+
+        if isinstance(arg.data, ir.BaseView):
```
Comment (drisspg) [inline torch/_inductor/lowering.py:2087]: why doesnt is_aligned_realized_tensor work here?

**Review by drisspg — APPROVED** [review]

File: torch/_inductor/lowering.py (hunk 1 above)
Comment (eellison) [reply torch/_inductor/lowering.py:2087]: These inputs don't necessarily have strides()

**Pull request closed without merging**